include/alpaka/mem/view/Accessor.hpp

```diff
@@ -0,0 +1,442 @@
+        {
+            return tex1Dfetch(texObj, z * slicePitchInValues + y * rowPitchInValues + x);
+        }
+
+        ALPAKA_FN_HOST_ACC auto operator()(float z, float y, float x) const -> TElem
+        {
+            return tex3D(texObj, x, y, z);
+        }
+
+        Image texObj;
+        TBufferIdx rowPitchInValues;
+        TBufferIdx slicePitchInValues;
+        Vec<DimInt<3>, TBufferIdx> extents;
+    };
+#endif
+
+    namespace internal
+    {
+        template<typename... TAccessModes>
+        struct BuildAccessModeList;
+
+        template<typename TAccessMode>
+        struct BuildAccessModeList<TAccessMode>
+        {
+            using type = TAccessMode;
+        };
+
+        template<typename TAccessMode1, typename TAccessMode2, typename... TAccessModes>
+        struct BuildAccessModeList<TAccessMode1, TAccessMode2, TAccessModes...>
+        {
+            using type = std::tuple<TAccessMode1, TAccessMode2, TAccessModes...>;
+        };
+
+        template<typename... TAccessModes, typename TBuf, std::size_t... TPitchIs, std::size_t... TExtentIs>
+        ALPAKA_FN_HOST_ACC auto buildAccessor(
+            TBuf&& buffer,
+            std::index_sequence<TPitchIs...>,
+            std::index_sequence<TExtentIs...>)
+        {
+            using DBuf = std::decay_t<TBuf>;
+            using TBufferIdx = Idx<DBuf>;
+            constexpr auto dim = Dim<DBuf>::value;
+            auto p = getPtrNative(buffer);
+            using AccessModeList = typename BuildAccessModeList<TAccessModes...>::type;
+            return Accessor<decltype(p), Elem<DBuf>, TBufferIdx, dim, AccessModeList>{
+                p,
+                getPitchBytes<TPitchIs + 1>(buffer)...,
+                {extent::getExtent<TExtentIs>(buffer)...}};
+        }
+
+        template<typename T>
+#ifdef __cpp_inline_variables
+        inline
+#endif
+            constexpr bool isAccessor
+            = false;
+
+        template<typename TMemoryHandle, typename TElem, typename TBufferIdx, std::size_t Dim, typename TAccessModes>
+#ifdef __cpp_inline_variables
+        inline
+#endif
+            constexpr bool isAccessor<Accessor<TMemoryHandle, TElem, TBufferIdx, Dim, TAccessModes>> = true;
+
+        template<typename List, typename Value>
+        struct MpContains : std::false_type
+        {
+        };
+
+        template<template<typename...> class List, typename Head, typename... Tail, typename Value>
+        struct MpContains<List<Head, Tail...>, Value>
+        {
+            static constexpr bool value = std::is_same<Head, Value>::value || MpContains<List<Tail...>, Value>::value;
+        };
+    } // namespace internal
+
+    //! Creates an accessor for the given memory object (view or buffer) using the specified access modes.
+    template<
+        typename... TAccessModes,
+        typename TBuf,
+        typename = std::enable_if_t<!internal::isAccessor<std::decay_t<TBuf>>>>
+    ALPAKA_FN_HOST_ACC auto accessWith(TBuf&& buffer)
```

Comment:
I think that this function (and its sister functions) should call type traits internally instead of some function hidden in `internal`. This would make it more consistent with alpaka. And with the current approach I can't add a specialization for SYCL buffers (or rather my alpaka wrapper around them) because a) partial function specialization is not allowed in C++ and b) they would fit the parameter list here and thus confuse the compiler with multiple valid overloads.

Reply:
Also: Why `accessWith` instead of `getAccess`or similar? What is the `With` referring to?

Reply:
With a glimpse at the code: we would need a customization point for `internal::buildAccessor`.

The with in `accessWith` referes to the non-deduced template parameter `TAccessMode`. So an example call would look like: `accessWith<ReadAccess>(memObject);`

Reply:
You should be able to specialize `BuildAccessor<sycl::buffer<...>>` now.